google/oauth2/credentials.py

```diff
@@ -254,6 +254,23 @@ def with_quota_project(self, quota_project_id):
             enable_reauth_refresh=self._enable_reauth_refresh,
         )
 
+    @_helpers.copy_docstring(credentials.CredentialsWithTokenUri)
```

Comment:
Why this is needed? What credentials types it affects?

Reply:
Python library uses helper methods to override things like quota project, scopes etc. checkout `with_quota_project` method in many credentials.
This override method is added to all the credentials that have a token uri.

Reply:
That is the question - do we need it for every credential type?

Reply:
Why the auth credetials has it as not implemented exception and here we have a default implementation? It is preferable to have explicit implementation

Reply:
I am following the existing pattern https://github.com/googleapis/google-auth-library-python/blob/main/google/auth/credentials.py#:~:text=class%20CredentialsWithQuotaProject(,quota%20project.%22)

Reply:
credentials.py is user credential. It is not a base class for other credentials. Base class is the credentials.py in auth package.

Reply:
figured out that this is in fact user creds